impossible-finance/New-Pair/src/agent.spec.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+  TransactionEvent,
+  createTransactionEvent,
+} from 'forta-agent';
+
+import { createAddress, TestTransactionEvent } from 'forta-agent-tools';
+
+import { provideHandleTransaction, SWAP_FACTORY_IFACE } from './agent';
+
+// Set the alertId to indicate it's a test
+const ALERT_ID: string = 'swap-factory-pair-created-test';
+
+// Address declarations 
+const factory: string = createAddress('0xa1');
+const token0: string = createAddress('0xb1');
+const token1: string = createAddress('0xb2');
+const pair: string = createAddress('0xc1');
+
+// Returns a finding with given inputs
+const createFinding = ( token0: string, token1: string, pair: string ) => Finding.fromObject({
+  name: 'New pair created',
+  description: 'A new pair has been created in Swap Factory V1',
+  alertId: ALERT_ID,
+  severity: FindingSeverity.Info,
+  type: FindingType.Info,
+  protocol: 'Impossible Finance',
+  metadata: {
+    token0: token0.toLowerCase(),
+    token1: token1.toLowerCase(), 
+    pair: pair.toLowerCase(),
+  }
+});
+
+describe('Swap-Factory-Pair-Created Agent test suite', () => {
+  const handler: HandleTransaction = provideHandleTransaction(
+    ALERT_ID,
+    factory,
+  );
+
+  it('should ignore empty transactions', async () => {
+    const tx: TransactionEvent = new TestTransactionEvent();
+
+    const findings = await handler(tx);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it('should ignore same events from other contracts', async () => {
+    const log = SWAP_FACTORY_IFACE.encodeEventLog(
+      SWAP_FACTORY_IFACE.getEvent('PairCreated'), [token0, token1, pair, '0']
+    );
+
+    const tx: TransactionEvent = new TestTransactionEvent().addAnonymousEventLog(
+      // Not the factory contract address
+      createAddress('0xd4'),
+      log.data,
+      ...log.topics
+    );
+
+    const findings: Finding[] = await handler(tx);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it('should ignore different event from same contract', async () => {
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .addEventLog(
+        'IrrelevantEvent(address)',
+        factory,
+        '',
+        token0,
+        token1,
+        SWAP_FACTORY_IFACE.getEventTopic('PairCreated'),
+      );
```

Comment:
Please at fill the event with data & topics accordingly with the signature you are providing

Reply:
This has been addressed in the latest commit